### PR TITLE
Gh 1336 sign tx with ledger

### PR DESCRIPTION
**Pull request by sche**
Handles #1336

### CODE REVIEW

File: Multisig/Logic/Ledger/LedgerController.swift [hunk 1]
```diff
@@ -30,14 +30,27 @@ class LedgerController {
...
         let command = getAddressCommand(path: path)
-        bluetoothController.sendCommand(device: device, command: command) { [weak self] result in
+
+        // We don't use [weak self] with private methods not to capture LedgerController in a caller
+        bluetoothController.sendCommand(device: device, command: command) { result in
             switch result {
             case .success(let data):
-                guard let addressString = self?.parseGetAddress(data: data),
+                guard data.count == 109,
+                      Int(data[5]) == 65, // public key length
```
Comment (elgatovital) [inline Multisig/Logic/Ledger/LedgerController.swift:39]: might be better to define constants with meaningful names for those values

File: Multisig/Logic/Ledger/LedgerController.swift [hunk 2]
```diff
@@ -30,14 +30,27 @@ class LedgerController {
         }
         let path = "44'/60'/0'/0/\(index)"
```
Comment (elgatovital) [inline Multisig/Logic/Ledger/LedgerController.swift:31]: do you account for different derivation paths? (ledger and ledger live)

File: Multisig/UI/Settings/OwnerKeyManagement/OnboardingLedgerKeyViewController.swift [hunk 3]
```diff
@@ -32,6 +32,29 @@ class OnboardingLedgerKeyViewController: AddKeyOnboardingViewController {
...
+
+extension OnboardingLedgerKeyViewController: SelectLedgerDeviceDelegate {
+    func selectLedgerDeviceViewController(_ controller: SelectLedgerDeviceViewController,
+                                          didSelectDevice deviceId: UUID,
+                                          bluetoothController: BluetoothController) {
+
+        let ledgerController = LedgerController(bluetoothController: bluetoothController)
+        ledgerController.getAddress(deviceId: deviceId, at: 10) { [weak controller] ledgerInfoOrNil in
+            guard let ledgerInfo = ledgerInfoOrNil else {
+                let alert = UIAlertController(title: "Address Not Found", message: "Please open Ethereum App on your Ledger device.", preferredStyle: .alert)
```
Comment (elgatovital) [inline Multisig/UI/Settings/OwnerKeyManagement/OnboardingLedgerKeyViewController.swift:48]: looks like hardcoded strings to me

**Review by elgatovital — APPROVED** [review]
tests are missing

File: Multisig/UI/Settings/OwnerKeyManagement/OnboardingLedgerKeyViewController.swift (hunk 3 above)
Comment (sche) [reply Multisig/UI/Settings/OwnerKeyManagement/OnboardingLedgerKeyViewController.swift:48]: yes, we use hardcoded strings everywhere

File: Multisig/Logic/Ledger/LedgerController.swift (hunk 2 above)
Comment (sche) [reply Multisig/Logic/Ledger/LedgerController.swift:31]: will be handled in future PRs

File: Multisig/Logic/Ledger/LedgerController.swift (hunk 1 above)
Comment (sche) [reply Multisig/Logic/Ledger/LedgerController.swift:39]: I think this is not necessary for low-level code and it is readable

**Comment by sche** [comment]
Create a separate issue for adding unit tests for LedgerController

**Pull request merged**